lldb/cmake/modules/LLDBConfig.cmake

```diff
@@ -56,6 +56,19 @@ macro(add_optional_dependency variable description package found)
   message(STATUS "${description}: ${${variable}}")
 endmacro()
 
+# BEGIN SWIFT MOD
+# Auto-detect whether we should use the static bindings.
+# This needs to happen before we detect any of the other dependencies because
+# their behavior relies on whether LLDB_USE_STATIC_BINDINGS is enabled or not.
+add_optional_dependency(LLDB_ENABLE_SWIG "Enable using SWIG to generate the LLDB bindings" SWIG SWIG_FOUND VERSION 3)
```

Comment:
I think this could probably be put on llvm.org. SWIG is technically an optional dependency since `LLDB_ENABLE_PYTHON` can be set to `OFF`. Same for Lua. Shouldn't be a blocker though.

Reply:
As is, I don't think it would be meaningful to put this upstream. There are no static bindings there. We could change the way we autodetect Python and Lua support though. Currently, we have `FindPythonAndSwig` and `FindLuaAndSwig` which only succeed if we find both. We could hoist finding SWIG out of those files, but we'd still have to check `LLDB_ENABLE_SWIG`. Is that what you're suggesting?

Reply:
Yes, something along those lines. I agree that this is not meaningful to put upstream in its current form. Upstream could be changed to support a `LLDB_ENABLE_SWIG` in the way you described, but I think that would be non-trivial work.

Reply:
https://reviews.llvm.org/D138879